python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -133,22 +147,36 @@ def runtimeEvalString(this_obj, attr_name, val):
 
 
 def raw_eval(val):
+    ## Attempts to evaluate the val object and return the object it evaluates to if it is a Python object
+    ## Makes use of basic caching as we don't expect that things at this level should change.
+    from copy import deepcopy
+    if val in _eval_cache:
+        return deepcopy(_eval_cache[val])
+
     try:
+        ## Catch common strings which shouldn't be evaluated anyway and return them asif an exception occured
+        if val in ['', '.', 'new']:
+            _eval_cache[val] = val
+            return deepcopy(val)
```

Comment:
without launching ```gannga --debug``` I don't know I've just  been reducing the total number of calls to this ass I want to avoid ever using it if possible